File: Paayaw_FuachieSobrehIncomeTaxCalculator.java
import java.util.Scanner;

public class Paayaw_FuachieSobrehIncomeTaxCalculator {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter your gross income. The program will print out your net income: ");
        double grossIncome = input.nextDouble();
        System.out.println(IncomeTax(grossIncome));  // Call the IncomeTax method and print the result
    }

    public static double IncomeTax(double grossIncome) {
        double firstBracket = 402.0;                    /* Initialization of brackets*/
        double secondBracket = 110.0;
        double thirdBracket = 130.0;
        double fourthBracket = 3000.0;
        double fifthBracket = 16395.0;
        double sixthBracket = 29963.0;
        double tax = 0;

        if (grossIncome <= firstBracket) {               /* For non taxable income*/
            return grossIncome;
        }

        double taxableAmount = grossIncome - firstBracket;           // Calculate the taxable amount by subtracting the first bracket

        if (taxableAmount >= secondBracket) {                  /* Checks if the taxable income is more than the bracket. If it is it updates tax*/
            tax = tax + 5.5;                                   /* If it is less than the bracket it updates tax by calculating the percentage*/
            taxableAmount = taxableAmount - secondBracket;
        } else if (taxableAmount < secondBracket) {
            tax = tax + (0.05 * taxableAmount);
            return grossIncome - tax;
        }

        if (taxableAmount >= thirdBracket) {
            tax = tax + 13;
            taxableAmount = taxableAmount - thirdBracket;
        } else if (taxableAmount < thirdBracket) {
            tax = tax + (0.1 * taxableAmount);
            return grossIncome - tax;
        }

        if (taxableAmount >= fourthBracket) {
            tax = tax + 525;
            taxableAmount = taxableAmount - fourthBracket;
        } else if (taxableAmount < fourthBracket) {
            tax = tax + (0.175 * taxableAmount);
            return grossIncome - tax;
        }

        if (taxableAmount >= fifthBracket) {
            tax = tax + 4098.75;
            taxableAmount = taxableAmount - fifthBracket;
        } else if (taxableAmount < fifthBracket) {
            tax = tax + (0.25 * taxableAmount);
            return grossIncome - tax;
        }

        if (taxableAmount >= sixthBracket) {
            tax = tax + 8988.90;
            taxableAmount = taxableAmount - sixthBracket;
        } else if (taxableAmount < sixthBracket) {
            tax = tax + (0.3 * taxableAmount);
            return grossIncome - tax;
        }

        if (taxableAmount > 0) {
            tax = tax + (0.35 * taxableAmount);
        }

        double netIncome = grossIncome - tax;

        return netIncome;
    }
}

File: Paayaw_FuachieSobrehIncomeTaxCalculator.py

def income_tax_calculator(n):
    first_bracket = 402.0                     # Initializing all the brackets
    second_bracket = 110.0
    third_bracket = 130.0
    fourth_bracket = 3000.0
    fifth_bracket = 16395.0
    sixth_bracket = 29963.0
    tax = 0

    if n <= first_bracket:                          # Checking for non-taxable income
        return n

    taxable_amount = n - first_bracket

    if taxable_amount >= second_bracket:                       # Checks if the taxable amount is greater than a bracket
        tax = tax + 5.5                                        # If the taxable income is more it updates the tax
        taxable_amount = taxable_amount - second_bracket       # If it's less calculates the percentage and updates tax

    elif taxable_amount < second_bracket:
        tax = tax + (0.05*taxable_amount)
        return n - tax

    if taxable_amount >= third_bracket:
        tax = tax + 13
        taxable_amount = taxable_amount - third_bracket

    elif taxable_amount < third_bracket:
        tax = tax + (0.1 * taxable_amount)
        return n - tax

    if taxable_amount >= fourth_bracket:
        tax = tax + 525
        taxable_amount = taxable_amount - fourth_bracket

    elif taxable_amount < fourth_bracket:
        tax = tax + (0.175 * taxable_amount)
        return n - tax

    if taxable_amount >= fifth_bracket:
        tax = tax + 4098.75
        taxable_amount = taxable_amount - fifth_bracket

    elif taxable_amount < fifth_bracket:
        tax = tax + (0.25 * taxable_amount)
        return n - tax

    if taxable_amount >= sixth_bracket:
        tax = tax + 8988.90
        taxable_amount = taxable_amount - sixth_bracket

    elif taxable_amount < sixth_bracket:
        tax = tax + (0.3 * taxable_amount)
        return n - tax

    if taxable_amount > 0:
        tax = tax + (0.35 * taxable_amount)

    net_income = n - tax

    return net_income


gross_income = int(input("Enter your gross income. The program will print out your net income. "))

print(income_tax_calculator(gross_income))
